### PR TITLE
cli: use kebab-case for --retry-tag-filter arg

### LINKED ISSUE #1291: cli: --retryTagFilter case inconsistent with other args
To specify a tag filter for retry:

```
$ cucumber-js --retryTagFilter @flaky
```

It being in camel case is inconsistent with others like `format-options`, `world-parameters` and `require-module` which are kebab-case.

This is not just aesthetic, as it means if you are using something like dargs to create profile strings on the fly, you can't rely on a consistent rule.

Suggest we change it to `--retry-tag-filter` - we could potentially keep the camel cased one as an alias for one major release cycle and make it output a warning for now? cc @charlierudolph

### PR DESCRIPTION
**Pull request by davidjgoss**
Fixes #1291 

- Issue a warning when used in camelCase (and make warnings from cli testable)
- Update the docs

### CODE REVIEW

File: src/cli/index.ts [hunk 1]
```diff
@@ -42,16 +42,33 @@ export default class Cli {
   private readonly argv: string[]
   private readonly cwd: string
   private readonly stdout: IFormatterStream
+  private readonly warn: (message: string) => void
 
-  constructor({ argv, cwd, stdout }) {
+  constructor({ argv, cwd, stdout, warn = console.warn }) {
```
Comment (charlierudolph) [inline src/cli/index.ts]: Could we pass in stderr and directly write to that (or create a [console](https://nodejs.org/api/console.html#console_new_console_stdout_stderr_ignoreerrors) given the passed in stdout / stderr)? I'd prefer that since we are already passing in stdout.

Comment (davidjgoss) [reply src/cli/index.ts]: Sure will look at that - I thought the `stdout` here was the output stream for the formatter

Comment (charlierudolph) [reply src/cli/index.ts]: Yeah, stdout is passed in for the formatter. The only reason to create our own console is to be able to still make use of the same API we were already using

Comment (davidjgoss) [reply src/cli/index.ts]: Done - creating a console from the stdout we pass in

Comment (charlierudolph) [reply src/cli/index.ts]: I was thinking we also pass in stderr and pass that into the console. The warning should definitely go to stderr. Is it currently going to stdout?

Comment (davidjgoss) [reply src/cli/index.ts]: Sorry yeah - going to `stderr` now

File: src/cli/index.ts [hunk 2]
```diff
@@ -42,16 +43,33 @@ export default class Cli {
...
   private readonly cwd: string
   private readonly stdout: IFormatterStream
+  private readonly console: Console
 
-  constructor({ argv, cwd, stdout }) {
+  constructor({ argv, cwd, stdout, stderr }) {
     this.argv = argv
     this.cwd = cwd
     this.stdout = stdout
+    this.console = new Console(stderr)
```
Comment (charlierudolph) [inline src/cli/index.ts]: I was thinking we pass both stdout and stderr to this. 

```typescript
this.console = new Console({stdout, stderr})
```

It's working currently due to the constructor assuming we are passing only a stdout stream and error output goes to stdout if stderr is not supplied. We usually wouldn't print something to stdout using the console variable but I think its possible I doubt we would even send anything to stdout using the console variable we are creating but I'm unsure.

More info here:

https://nodejs.org/api/console.html#console_new_console_stdout_stderr_ignoreerrors

File: src/cli/index.ts [hunk 3]
```diff
@@ -42,16 +43,33 @@ export default class Cli {
...
+      cwd: this.cwd,
+    })
+  }
+
+  private lintArgv(fullArgv: string[]): void {
+    if (fullArgv.includes('--retryTagFilter')) {
+      this.console.warn(
+        'the argument --retryTagFilter is deprecated and will be removed in a future release; please use --retry-tag-filter'
+      )
+    }
```
Comment (charlierudolph) [inline src/cli/index.ts]: Could we put this in the argvParser? 

If we do that, I'm uncertain its worth trying to plumb through stderr to there, but I'm also not opposed to that. Alternatively, we could just use `@spawn` in the feature test

File: src/cli/index.ts (hunk 2 above)
Comment (davidjgoss) [reply src/cli/index.ts]: Gotcha, fixed

File: src/cli/index.ts (hunk 3 above)
Comment (davidjgoss) [reply src/cli/index.ts]: I did have it as part of the `parse` method there originally, but the warning would be logged twice if you used the camelCased version from the command line, because `parse` is invoked once with just the argv from the command line, then again with the ones from the resolved profile prepended.

Comment (charlierudolph) [reply src/cli/index.ts]: We could add an option to the argvParser to disable linting for when we are loading just the profile

Comment (davidjgoss) [reply src/cli/index.ts]: Made some changes, now in a `lint` method on the parser which is called at the right moment, and using `@spawn` in the scenario so we don't have to create a console and pass it around.

File: features/support/world.ts [hunk 4]
```diff
@@ -60,25 +60,30 @@ export class World {
       })
     } else {
       const stdout = new PassThrough()
+      const stderr = new PassThrough()
```
Comment (charlierudolph) [inline features/support/world.ts]: Are these changes still needed? Appears its simpler to not have them

Comment (davidjgoss) [reply features/support/world.ts]: True, removed now
